component/org.wso2.carbon.identity.oauth2.token.handler.clientauth.mutualtls/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/mutualtls/MutualTLSClientAuthenticator.java

```diff
@@ -119,17 +116,32 @@ public boolean authenticateClient(HttpServletRequest request, Map<String, List>
             }
             X509Certificate requestCert;
             Object certObject = request.getAttribute(JAVAX_SERVLET_REQUEST_CERTIFICATE);
-            if (certObject instanceof X509Certificate[]) {
-                X509Certificate[] cert = (X509Certificate[]) certObject;
-                requestCert = cert[0];
-            } else if (certObject instanceof X509Certificate){
-                requestCert = (X509Certificate) certObject;
+            if (certObject != null) {
```

Comment:
instanceOf anyway checks for null. We can add another else if condition here to check whether the certificate from header is available.

Reply:
Fixed https://github.com/wso2-extensions/identity-oauth-addons/pull/54/commits/c65eaebc9ee402ead6b0f2dc17e84de4738b2d8c